.github/workflows/main.yml

```diff
@@ -48,4 +48,24 @@ jobs:
       - name: Run basic yaml validation
         run: |
           python tests/validate_yaml.py
-        
+  on-fail:
+    if: failure()
+    runs-on: ubuntu-latest
+    needs: check
+    steps:
+      - name: Add label to PR
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const label = 'staging-skip';
+            const { context, github } = require('@actions/github');
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+            const pull_number = context.payload.pull_request.number;
```

Comment:
This action runs also on [direct pushes to `main`](https://github.com/kernelci/kernelci-pipeline/pull/712/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3R9).

Adding a `staging-skip` label wouldn't make much sense in that context (there's no PR) but should this case be also covered somehow if any of the validation steps fails on such a push?

Reply:
fixed and set only on pull_request